uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/ZoneControllerIntegrationTests.java

```diff
@@ -0,0 +1,160 @@
+
+import static org.cloudfoundry.identity.uaa.zone.ZoneService.X_IDENTITY_ZONE_ID;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.UUID;
+
+import org.cloudfoundry.identity.uaa.ServerRunning;
+import org.cloudfoundry.identity.uaa.test.TestAccountSetup;
+import org.cloudfoundry.identity.uaa.test.UaaTestAccounts;
+import org.cloudfoundry.identity.uaa.zone.IdentityZone;
+import org.cloudfoundry.identity.uaa.zone.model.ZoneResponse;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.client.ClientHttpResponse;
+import org.springframework.security.oauth2.client.OAuth2RestTemplate;
+import org.springframework.security.oauth2.client.http.OAuth2ErrorHandler;
+import org.springframework.security.oauth2.client.test.OAuth2ContextConfiguration;
+import org.springframework.security.oauth2.client.test.OAuth2ContextSetup;
+import org.springframework.security.oauth2.client.token.grant.client.ClientCredentialsResourceDetails;
+import org.springframework.web.client.RestTemplate;
+
+@OAuth2ContextConfiguration(ZoneControllerIntegrationTests.ZoneClient.class)
+public class ZoneControllerIntegrationTests {
+
+    @Rule
+    public ServerRunning serverRunning = ServerRunning.isRunning();
+
+    private final UaaTestAccounts testAccounts = UaaTestAccounts.standard(serverRunning);
+
+    @Rule
+    public OAuth2ContextSetup context = OAuth2ContextSetup.standard(serverRunning);
+
+    @Rule
+    public TestAccountSetup testAccountSetup = TestAccountSetup.standard(serverRunning, testAccounts);
+
+    private RestTemplate client;
+
+    @Before
+    public void createRestTemplate() {
+        client = (OAuth2RestTemplate) serverRunning.getRestTemplate();
+        client.setErrorHandler(new OAuth2ErrorHandler(context.getResource()) {
+            // Pass errors through in response entity for status code analysis
+            @Override
+            public boolean hasError(ClientHttpResponse response) {
+                return false;
+            }
+
+            @Override
+            public void handleError(ClientHttpResponse response) {
+            }
+        });
+    }
+
+    @Test
+    public void testGetZone() {
+        //TODO: delete once the orchestrator create API implemented
+        IdentityZone identityZone = createZone();
+        assertNotNull(identityZone);
+        ResponseEntity<ZoneResponse> response = client.getForEntity(
+            serverRunning.getUrl("/zones") + "?name=" + identityZone.getName(),
+            ZoneResponse.class);
+        if (response.getStatusCode().is2xxSuccessful()) {
```

Comment:
Please remove if condition from here also like we did for mock movc tests.

Reply:
I think Mockmvc we are checking the status while fetch the request itself (andExpect(status().isNotFound())), but here we are not checking. So I use this condition to check. Probably I will add the fail assertion in else.